### PR TITLE
Sort the user provided env for rex

### PR DESCRIPTION
**Pull request by Tatskaari**

### CODE REVIEW

**Comment by Tatskaari** [comment]
I think this might also contribute to #2263

**Comment by tobyfielding1** [comment]
noice!

File: src/sandbox/sandbox_linux.go [hunk 1]
```diff
@@ -68,7 +68,10 @@ func Sandbox(args []string) error {
 		}
 	}
 	err = syscall.Exec(cmd, args, env)
-	return fmt.Errorf("Failed to exec %s: %s", cmd, err)
+	if err != nil {
+		return fmt.Errorf("Failed to exec %s: %s", cmd, err)
+	}
+	return nil
```
Comment (Tatskaari) [inline src/sandbox/sandbox_linux.go:74]: not sure how this got in here :disappointed:

Comment (peterebden) [reply src/sandbox/sandbox_linux.go:74]: Guessing there's a newer linter now that didn't used to check this.

I think it's wrong in this case though, `Exec` only ever returns on failure so the code before was legit.

**Review by peterebden — APPROVED** [review]
Can't wait for generics so we can have ordered maps...

File: src/sandbox/sandbox_linux.go (hunk 1 above)
Comment (Tatskaari) [reply src/sandbox/sandbox_linux.go:74]: I guess the linter doesn't know that. This shuts it up so I think it's fine.

Comment (peterebden) [reply src/sandbox/sandbox_linux.go:74]: Yeah IIRC there are ways of marking this kind of thing in C compilers but it's not something Go knows about. I guess it's not a big deal, although maybe that linter could special-case one or two functions like this.

**Comment by Tatskaari** [comment]
+1 on generics... will be great to have some proper data types.

**Pull request merged**